public/app/features/plugins/admin/hooks/usePluginDetailsTabs.tsx

```diff
@@ -1,104 +1,144 @@
-import { useMemo } from 'react';
+import React, { useMemo } from 'react';
 import { useLocation } from 'react-router-dom';
 
-import { PluginIncludeType, PluginType } from '@grafana/data';
+import { GrafanaPlugin, NavModelItem, PluginIncludeType, PluginType } from '@grafana/data';
 import { config } from '@grafana/runtime';
+import { toIconName } from '@grafana/ui';
 import { contextSrv } from 'app/core/core';
 import { AccessControlAction } from 'app/types';
 
+import { PluginDetailsHeader } from '../components/PluginDetailsHeader';
 import { usePluginConfig } from '../hooks/usePluginConfig';
 import { isOrgAdmin } from '../permissions';
-import { CatalogPlugin, PluginDetailsTab, PluginTabIds, PluginTabLabels } from '../types';
+import { CatalogPlugin, PluginTabIds, PluginTabLabels } from '../types';
 
 type ReturnType = {
   error: Error | undefined;
   loading: boolean;
-  tabs: PluginDetailsTab[];
-  defaultTab: string;
+  navModel: NavModelItem;
+  activePageId: PluginTabIds | string;
 };
 
-export const usePluginDetailsTabs = (plugin?: CatalogPlugin, defaultTabs: PluginDetailsTab[] = []): ReturnType => {
+export const usePluginDetailsTabs = (plugin?: CatalogPlugin, pageId?: PluginTabIds): ReturnType => {
   const { loading, error, value: pluginConfig } = usePluginConfig(plugin);
-  const isPublished = Boolean(plugin?.isPublished);
   const { pathname } = useLocation();
+  const defaultTab = useDefaultPage(plugin, pluginConfig);
+  const parentUrl = pathname.substring(0, pathname.lastIndexOf('/'));
+  const isPublished = Boolean(plugin?.isPublished);
 
-  const [tabs, defaultTab] = useMemo(() => {
+  const currentPageId = pageId || defaultTab;
+  const navModelChildren = useMemo(() => {
     const canConfigurePlugins =
       plugin && contextSrv.hasAccessInMetadata(AccessControlAction.PluginsWrite, plugin, isOrgAdmin());
-    const tabs: PluginDetailsTab[] = [...defaultTabs];
-    let defaultTab;
+    const navModelChildren: NavModelItem[] = [];
     if (isPublished) {
-      tabs.push({
-        label: PluginTabLabels.VERSIONS,
-        icon: 'history',
+      navModelChildren.push({
+        text: PluginTabLabels.VERSIONS,
         id: PluginTabIds.VERSIONS,
-        href: `${pathname}?page=${PluginTabIds.VERSIONS}`,
+        icon: toIconName('history'),
```

Comment:
I think we can remove the `toIconName` and use the valid IconName here directly.

Reply:
Ah thanks for spotting that. Have updated it. 👍